README.md
# Pinto WordPress Theme

Pinto is a delicate & delightfully lightweight theme built for writers, journalists & photographers. It's a seductively minimal WordPress theme that takes just minutes to get running.

## Development

1. Clone this GitHub repository.
2. Browse to the folder in the command line.
3. Run the `npm install` command to install the theme's development dependencies within a /node_modules/ folder.
4. Add an environment.json to the theme root directory with a key of `devURL` and a value of your local install.

Example: 
```
{
    "devURL": "http://demo.dev/pinto"
}
```
4. Run the `npm start` command for development.
5. Run the `build` gulp task to process build files and generate a zip.

## Support

[GoDaddy has acquired ThemeBeans](https://richtabor.com/?p=907), and in the spirit of open source — the entire ThemeBeans WordPress theme catalog is now freely available here on GitHub. However, **theme support and remote updates are only available for current license holders**, and up to April 9th, 2020.


changelog.txt
** Pinto Changelog **

05 April 2019, Version 3.0.8
	* New: Add support for the new `wp_body_open` hook in WordPress 5.2+
	* Fix: Adjust Alert block typography control functionality
	* Tweak: Add proper styling for the new <code> format within the block editor
	* Tweak: Admin 1.6.1

26 February 2019, Version 3.0.7
	* New: Add support for the upcoming CoBlocks rounded and 3D button styles
	* Fix: Multiple paragraphs within a Quote block are now properly spaced
	* Fix: Adjust Alert block typography control functionality
	* Tweak: Adjust core Button block styling

30 January 2019, Version 3.0.6
	* Fix: <strong> tags now render properly on the frontend

28 January 2019, Version 3.0.5
	* Tweak: Custom fonts applied with CoBlocks now render properly on blockquotes
	* Fix: Mobile menu toggle now behaves appropriately on mobile
	* Fix: Improve how the video post format metabox renders previews
	* Fix: Adjust video post format iFrame

27 January 2019, Version 3.0.4
	* Fix: Resolve Webkit button appearance issue

26 January 2019, Version 3.0.3
	* Fix: Resolve issue where comments did not always appear on new posts

06 January 2019, Version 3.0.2
	* Tweak: Improve caption link style
	* Tweak: Add appropriate blockquote fallback
	* Tweak: Improve metabox styling within Gutenberg

06 December 2018, Version 3.0.1
	* Tweak: Hidden sidebar/mobile menu animation speed
	* Tweak: Adjust pullquote styling

05 December 2018, Version 3.0.0
	* New: Full support for Gutenberg and WordPress 5.0
	* New: Apply massive framework improvment, which is now based on Underscores
	* New: Add support for CoBlocks, Gutenberg Blocks for Content Marketers - https://coblocks.com
	* New: Add support for Block Gallery, gallery Gutenberg Blocks - https://wpblockgallery.com
	* New: Add support for the Merlin WP theme setup wizard 🧙‍♂️
	* New: Add custom Gutenberg editor color palette
	* New: Add selected Customizer accent color to the Gutenberg editor color palette
	* New: Add custom editor font sizes
	* New: Add support for PHP 7.2
	* New: Add TGM Plugin Activation support
	* New: Add concated styles and scripts for faster performance 🚀
	* New: Add support for core custom logo functionality and auto-migration
	* New: Add SCSS support with auto browser auto-prefixing
	* New: Add minified Customizer JS and editor stylesheet assets with auto-prefixing
	* New: Add new custom logo width options for desktop and mobile viewports
	* New: Add new Underscore JS Customizer controls for improved Customizer speed
	* New: Add better customizer defaults for improved onboarding
	* New: Add new Customizer based theme license activator
	* New: Add new debug mode that works with minified asset delivery throughout the entire theme
	* Tweak: Improve Customizer sanitization functions
	* Tweak: Conducted a security audit and added recommended optimizations
	* Tweak: Remove single post thumbnails
	* Tweak: Update comments
	* Tweak: Apply modern styling
	* Tweak: Update child theme
	* Remove: Legacy framework code
	* Remove: Archive and Site Map templates, as they are now provided by Gutenberg
	* Remove: Gallery post format, as it is now provided by Gutenberg & Block Gallery

8 December 2016 - Version 2.9.1
    Fixes & Updates:
        * WordPress 4.7 touchups
        * Custom CSS -> WP Core CSS migration
        * Updated pingback header
        * Updated languages

29 February 2016 - Version 2.9.0
    * Admin favicon fix
    * EDD_Theme_Updater Update
        - You will need to re-activate your theme license.
    * Updated languages

04 January 2016 - Version 2.8.1
	* Added Grunt task runner
	* Dynamic Sidebar WP 4.4 fix
		- content-hidden-sidebar.php
	* Updated languages

2 September 2015 - Version 2.8.0
	* 100% GPL license
		- style.css (top comments)
		- Read more: http://themebeans.com/news/all-themes-now-100-gpl/
	* Theme Customizer update
		- Moved Branding section to WP default Site Identity section
		- Removed favicon & apple touch uploader (WP 4.3 is default)
		- inc/customizer folder
	* 4.3 widget __construct update
		- inc/widgets folder
	* Framework updated to v2.6.2.1
		- WP 4.3 Theme Customizer update
		- Custom CSS Editor UI/X update
		- framework folder
	* Updated languages

30 April 2015 - Version 2.7.0
	* XSS fix
	* Visual post editor touch ups
	* Theme Customizer UI improvements
	* bean_custom_form_filters comments 4.2+ fix
	* Table of Contents for primary stylesheet
	* Framework updated to v2.2.5
		- Updated Theme Customizer UI
		- Updated Custom CSS Editor
	* Updated languages

15 January 2015 - Version 2.6.2
	* Contact form updated
		- Anti-spam $headers fix
	* Framework updated to v2.2.3.1
		- Customizer image upload update
	* Updated languages

22 December 2014 - Version 2.6.1
	* Framework updated to v2.2.1
	* Updated languages

9 December 2014 - Version 2.6
	* Framework updated to v2.2
		- Updated Custom CSS Editor
		- Notifications for theme updates
		- Dashboard updating enabled
		- Theme license admin page (NEW)
		- Moved customizer functions to /framework
	* Updated languages

1 October 2014 - Version 2.5.0
	* Animation engine update
	* Framework updated to v2.1.1
		- Customizer major update
	* Updated languages

5 September 2014 - Version 2.4.0
	* WP 4.0 compatibility update
	* Added site map template (NEW)
	* Framework updated to v2.0.9.1
	* Modded 'page' filenames to 'template'
	* Updated styles for templates' mod
	* Moved framework.css into style.css - performance improvement
	* Removed framework.css
	* Updated languages

30 June 2014 - Version 2.3.0
	* WP 3.9+ UI touch ups
	* Added video embed widget (NEW)
	* Added fallback/function for more tag
	* Support for Bean 500px plugin
	* Framework updated to v2.0.9
	* Updated languages

16 Mar 2014 - Version 2.2.0
	* New theme version notifications/update system
	* New Custom CSS Editor
	* Theme customizer UI update
	* Framework updated to v2.0.5
	* Updated languages

12 Feb 2014 - Version 2.1.0
	* Support for Bean Shortcodes 2.0+

22 Jan 2014 - Version 2.0.2

functions.php
<?php
/**
 * Theme functions and definitions
 *
 * @link https://developer.wordpress.org/themes/basics/theme-functions/
 *
 * @package     Pinto
 * @link        https://themebeans.com/themes/pinto
 */

if ( ! defined( 'PINTO_DEBUG' ) ) :
	/**
	 * Check to see if development mode is active.
	 * If set to false, the theme will load un-minified assets.
	 */
	define( 'PINTO_DEBUG', true );
endif;

if ( ! defined( 'PINTO_ASSET_SUFFIX' ) ) :
	/**
	 * If not set to true, let's serve minified .css and .js assets.
	 * Don't modify this, unless you know what you're doing!
	 */
	if ( ! defined( 'PINTO_DEBUG' ) || true === PINTO_DEBUG ) {
		define( 'PINTO_ASSET_SUFFIX', null );
	} else {
		define( 'PINTO_ASSET_SUFFIX', '.min' );
	}
endif;

/**
 * Sets up theme defaults and registers support for various WordPress features.
 *
 * Note that this is hooked into the after_setup_theme hook, which
 * runs before the init hook. The init hook is too late for some features, such
 * as indicating support for post thumbnails.
 */
function pinto_setup() {
	/*
	 * Make theme available for translation.
	 * Translations can be filed in the /languages/ directory.
	 * If you're building a theme based on Tabor, use a find and replace
	 * to change 'pinto' to the name of your theme in all the template files
	 */
	load_theme_textdomain( 'pinto', get_parent_theme_file_path( '/languages' ) );

	/*
	 * Add default posts and comments RSS feed links to head.
	 */
	add_theme_support( 'automatic-feed-links' );

	/*
	 * Let WordPress manage the document title.
	 * By adding theme support, we declare that this theme does not use a
	 * hard-coded <title> tag in the document head, and expect WordPress to
	 * provide it for us.
	 */
	add_theme_support( 'title-tag' );

	/*
	 * Enable support for Post Thumbnails on posts and pages.
	 *
	 * @link http://codex.wordpress.org/Function_Reference/add_theme_support#Post_Thumbnails
	 */
	add_theme_support( 'post-thumbnails' );

	set_post_thumbnail_size( 140, 140 );
	add_image_size( 'thumbnail-large', 2000, 9999 );
	add_image_size( 'thumbnail-small', 250, 9999 );

	/*
	 * This theme uses wp_nav_menu() in the following locations.
	 */
	register_nav_menus(
		array(
			'primary-menu' => esc_html__( 'Primary Navigation', 'pinto' ),
		)
	);

	// Set up the WordPress core custom background feature.
	add_theme_support(
		'custom-background',
		apply_filters(
			'pinto_custom_background_args',
			array(
				'default-color' => 'ffffff',
				'default-image' => '',
			)
		)
	);

	/*
	 * Enable support for Post Formats.
	 *
	 * See: https://codex.wordpress.org/Post_Formats
	 */
	add_theme_support(
		'post-formats', array(
			'video',
		)
	);

	/*
	 * Switch default core taborup for search form, comment form, and comments
	 * to output valid HTML5.
	 */
	add_theme_support(
		'html5', array(
			'search-form',
			'comment-form',
			'comment-list',
			'gallery',
			'caption',
		)
	);

	/*
	 * Enable support for the WordPress default Theme Logo
	 * See: https://codex.wordpress.org/Theme_Logo
	 */
	add_theme_support(
		'custom-logo', array(
			'flex-width' => true,
		)
	);

	/*
	 * Enable support for Customizer Selective Refresh.
	 * See: https://make.wordpress.org/core/2016/02/16/selective-refresh-in-the-customizer/
	 */
	add_theme_support( 'customize-selective-refresh-widgets' );

	/*
	 * Enable support for responsive embedded content
	 * See: https://wordpress.org/gutenberg/handbook/extensibility/theme-support/#responsive-embedded-content
	 */
	add_theme_support( 'responsive-embeds' );

	/**
	 * Custom colors for use in the editor.
	 *
	 * @link https://wordpress.org/gutenberg/handbook/reference/theme-support/
	 */
	add_theme_support(
		'editor-color-palette', array(
			array(
				'name'  => esc_html__( 'Black', 'pinto' ),
				'slug'  => 'black',
				'color' => '#2a2a2a',
			),
			array(
				'name'  => esc_html__( 'Gray', 'pinto' ),
				'slug'  => 'gray',
				'color' => '#727477',
			),
			array(
				'name'  => esc_html__( 'Light Gray', 'pinto' ),
				'slug'  => 'light-gray',
				'color' => '#f8f8f8',
			),
			array(
				'name'  => esc_html__( 'White', 'pinto' ),
				'slug'  => 'white',
				'color' => '#ffffff',
			),
			array(
				'name'  => esc_html__( 'Titan White', 'pinto' ),
				'slug'  => 'titan-white',
				'color' => '#E0D8E2',
			),
			array(
				'name'  => esc_html__( 'Tropical Blue', 'pinto' ),
				'slug'  => 'tropical-blue',
				'color' => '#C5DCF3',
			),
			array(
				'name'  => esc_html__( 'Peppermint', 'pinto' ),
				'slug'  => 'peppermint',
				'color' => '#d0eac4',
			),
			array(
				'name'  => esc_html__( 'Iceberg', 'pinto' ),
				'slug'  => 'iceberg',
				'color' => '#D6EFEE',
			),
			array(
				'name'  => esc_html__( 'Bridesmaid', 'pinto' ),
				'slug'  => 'bridesmaid',
				'color' => '#FBE7DD',
			),
			array(
				'name'  => esc_html__( 'Pipi', 'pinto' ),
				'slug'  => 'pipi',
				'color' => '#fbf3d6',
			),
			array(
				'name'  => esc_html__( 'Accent', 'pinto' ),
				'slug'  => 'accent',
				'color' => esc_html( get_theme_mod( 'theme_accent_color', pinto_defaults( 'theme_accent_color' ) ) ),
			),
		)
	);

	/**
	 * Custom font sizes for use in the editor.
	 *
	 * @link https://wordpress.org/gutenberg/handbook/extensibility/theme-support/#block-font-sizes
	 */
	add_theme_support(
		'editor-font-sizes', array(
			array(
				'name'      => esc_html__( 'Small', 'pinto' ),
				'shortName' => esc_html__( 'S', 'pinto' ),
				'size'      => 16,
				'slug'      => 'small',
			),
			array(
				'name'      => esc_html__( 'Medium', 'pinto' ),
				'shortName' => esc_html__( 'M', 'pinto' ),
				'size'      => 19,
				'slug'      => 'medium',
			),
			array(
				'name'      => esc_html__( 'Large', 'pinto' ),
				'shortName' => esc_html__( 'L', 'pinto' ),
				'size'      => 24,
				'slug'      => 'large',
			),
			array(
				'name'      => esc_html__( 'Huge', 'pinto' ),
				'shortName' => esc_html__( 'XL', 'pinto' ),
				'size'      => 30,
				'slug'      => 'huge',
			),
		)
	);

	// Add support for default block styles.
	add_theme_support( 'wp-block-styles' );

	// Add support for full and wide align images.
	add_theme_support( 'align-wide' );

	// Add support for editor styles.
	add_theme_support( 'editor-styles' );

	// Enqueue editor styles.
	add_editor_style( 'assets/css/style-editor' . PINTO_ASSET_SUFFIX . '.css' );

	// Enqueue fonts in the editor.
	add_editor_style( pinto_fonts_url() );

}
add_action( 'after_setup_theme', 'pinto_setup' );

/**
 * Set the content width in pixels, based on the theme's design and stylesheet.
 *
 * Priority 0 to make it available to lower priority callbacks.
 *
 * @global int $content_width
 */
function pinto_content_width() {
	$GLOBALS['content_width'] = apply_filters( 'pinto_content_width', 660 );
}
add_action( 'after_setup_theme', 'pinto_content_width', 0 );

/**
 * Register widget areas.
 *
 * @link http://codex.wordpress.org/Function_Reference/register_sidebar
 */
function pinto_widgets_init() {

	register_sidebar(
		array(
			'name'          => esc_html__( 'Hidden Sidebar', 'pinto' ),
			'id'            => 'internal-sidebar',
			'description'   => esc_html__( 'Widget area for the hidden sidebar.', 'pinto' ),
			'before_widget' => '<div class="widget %2$s clearfix">',
			'after_widget'  => '</div>',
			'before_title'  => '<h5 class="widget-title h4">',
			'after_title'   => '</h5>',
		)
	);
}
add_action( 'widgets_init', 'pinto_widgets_init' );

/**
 * Enqueue scripts and styles.
 */
function pinto_scripts() {

	// Add custom fonts, used in the main stylesheet.
	wp_enqueue_style( 'pinto-fonts', pinto_fonts_url(), false, '@@pkg.version', 'all' );

	// Load theme styles.
	if ( is_child_theme() ) {
		wp_enqueue_style( 'pinto-style', get_parent_theme_file_uri( '/style' . PINTO_ASSET_SUFFIX . '.css' ), false, '@@pkg.version' );
		wp_enqueue_style( 'pinto-child-style', get_theme_file_uri( '/style.css' ), false, '@@pkg.version', 'all' );
	} else {
		wp_enqueue_style( 'pinto-style', get_theme_file_uri( '/style' . PINTO_ASSET_SUFFIX . '.css' ), false, '@@pkg.versio' );
	}

	/**
	 * Now let's check the same for the scripts.
	 */
	if ( SCRIPT_DEBUG || PINTO_DEBUG ) {

		// Vendor scripts.
		wp_enqueue_script( 'pinto-libraries', get_theme_file_uri( '/assets/js/vendors/custom-libraries.js' ), array( 'jquery' ), '@@pkg.version', true );
		wp_enqueue_script( 'jquery.waypoints', get_theme_file_uri( '/assets/js/vendors/waypoints-sticky.min.js' ), array( 'jquery' ), '@@pkg.version', true );
		wp_enqueue_script( 'jquery.superfish', get_theme_file_uri( '/assets/js/vendors/superfish.js' ), array( 'jquery' ), '@@pkg.version', true );

		// Custom script.
		wp_enqueue_script( 'pinto-global', get_theme_file_uri( '/assets/js/custom/global.js' ), array( 'jquery' ), '@@pkg.version', true );

		$translation_handle = 'pinto-global'; // Variable for wp_localize_script.

	} else {
		wp_enqueue_script( 'pinto-vendors-min', get_theme_file_uri( '/assets/js/vendors.min.js' ), array( 'jquery' ), '@@pkg.version', true );
		wp_enqueue_script( 'pinto-custom-min', get_theme_file_uri( '/assets/js/custom.min.js' ), array( 'jquery' ), '@@pkg.version', true );

		$translation_handle = 'pinto-custom-min'; // Variable for wp_localize_script for minified javascript.
	}

	// Load the standard WordPress comments reply javascript.
	if ( is_singular( 'post' ) && comments_open() && get_option( 'thread_comments' ) ) {
		wp_enqueue_script( 'comment-reply' );
	}
}
add_action( 'wp_enqueue_scripts', 'pinto_scripts' );

/**
 * Remove the duplicate stylesheet enqueue for older versions of the child theme.
 *
 * Since v3.0.0 Pinto has a built-in auto-loader for loading the appropriate
 * parent theme stylesheet, without the need for a wp_enqueue_scripts function within
 * the child theme. This means that stylesheets will "just work" and there's less chance
 * that users will accidently disrupt stylesheet loading.
 */
function pinto_remove_duplicate_child_parent_enqueue_scripts() {
	remove_action( 'wp_enqueue_scripts', 'pinto_child_scripts', 10 );
}
add_action( 'init', 'pinto_remove_duplicate_child_parent_enqueue_scripts' );

/**
 * Register custom fonts.
 */
function pinto_fonts_url() {
	$fonts_url = '';

	/*
	 * Translators: If there are characters in your language that are not
	 * supported by Plex, translate this to 'off'. Do not translate
	 * into your own language.
	 */
	$plex = esc_html_x( 'on', 'Plex font: on or off', 'pinto' );

	/*
	 * Translators: If there are characters in your language that are not
	 * supported by PT Serif, translate this to 'off'. Do not translate
	 * into your own language.
	 */
	$pt_serif = esc_html_x( 'on', 'PT Serif font: on or off', 'pinto' );

	if ( 'off' !== $plex || 'off' !== $pt_serif ) {
		$font_families = array();

		if ( 'off' !== $plex ) {
			$font_families[] = 'IBM Plex Sans:400,700i';
		}

		if ( 'off' !== $pt_serif ) {
			$font_families[] = 'PT Serif:400,700,400italic';
		}

		$query_args = array(
			'family' => rawurlencode( implode( '|', $font_families ) ),
			'subset' => rawurlencode( 'latin,latin-ext' ),
		);

		$fonts_url = add_query_arg( $query_args, 'https://fonts.googleapis.com/css' );
	}

	return esc_url_raw( $fonts_url );
}

/**
 * Add preconnect for Google Fonts.
 *
 * @param  array|array   $urls           URLs to print for resource hints.
 * @param  string|string $relation_type  The relation type the URLs are printed.
 * @return array|array   $urls           URLs to print for resource hints.
 */
function pinto_resource_hints( $urls, $relation_type ) {
	if ( wp_style_is( 'pinto-fonts', 'queue' ) && 'preconnect' === $relation_type ) {
		$urls[] = array(
			'href' => 'https://fonts.gstatic.com',
			'crossorigin',
		);
	}

	return $urls;
}
add_filter( 'wp_resource_hints', 'pinto_resource_hints', 10, 2 );

/**
 * Add a pingback url auto-discovery header for singularly identifiable articles.
 */
function pinto_pingback_header() {
	if ( is_singular() && pings_open() ) {
		echo '<link rel="pingback" href="', bloginfo( 'pingback_url' ), '">';
	}
}

add_action( 'wp_head', 'pinto_pingback_header' );

/**
 * Manually set an excerpt length.
 */
function pinto_custom_excerpt_length( $length ) {
	return 34;
}
add_filter( 'excerpt_length', 'pinto_custom_excerpt_length', 999 );

/**
 * Custom template tags for this theme.
 */
require get_theme_file_path( '/inc/template-tags.php' );

/**
 * Metaboxes.
 */
require get_theme_file_path( '/inc/metaboxes.php' );

/**
 * Widgets.
 */
require get_theme_file_path( '/inc/widgets/widget-flickr.php' );
require get_theme_file_path( '/inc/widgets/widget-newsletter.php' );

/**
 * Customizer additions.
 */
require get_theme_file_path( '/inc/customizer/customizer.php' );
require get_theme_file_path( '/inc/customizer/defaults.php' );
require get_theme_file_path( '/inc/customizer/customizer-css.php' );
require get_theme_file_path( '/inc/customizer/customizer-editor.php' );
require get_theme_file_path( '/inc/customizer/sanitization.php' );

/**
 * Admin specific functions.
 */
require get_parent_theme_file_path( '/inc/admin/init.php' );

/**
 * Disable Dashboard Doc.
 */
function themebeans_guide() {}
